client/src/boot/buildApolloClient.js

```diff
@@ -1,53 +1,87 @@
-import ApolloClient, { createNetworkInterface } from 'apollo-client';
-import { printRequest } from 'apollo-client/transport/networkInterface';
+/* global window */
+import fetch from 'isomorphic-fetch';
+import ApolloClient, { InMemoryCache, IntrospectionFragmentMatcher, printAST, ApolloLink } from 'apollo-boost';
+import { BatchHttpLink } from 'apollo-link-batch-http';
 import qs from 'qs';
 
 function buildApolloClient(baseUrl) {
```

Comment:
It might be worthwhile splitting some of that logic up and cutting down a bit on the inline variables/function. It would make it easier to figure out what's going on and allow us lesser mortals to follow along.

`buildApolloClient` could become something like:
```JS
function buildApolloClient(baseUrl) {
    return fetch(`${baseUrl}graphql`, httpOptions)
       .then(doSomething)
       .then(doSomethingElse)
       .then(doFinalThing)
}
```
Each method applied to the promise should have proper doc.

On another note, should there be some error handling on `fetch`? It's an HTTP call, so there's a chance it will fail. Not sure what the best practice is here.

Reply:
Kind of a coincidence, but maybe not so much -- this is exactly the thing I noticed when I was reviewing it, and refactored this into smaller bits as you suggested. I just haven't pushed up the branch yet.